File: android/app/src/main/java/com/scholarlyapps/pathlingo/data/local/TokenStore.java
package com.scholarlyapps.pathlingo.data.local;

import android.content.Context;
import android.content.SharedPreferences;

public class TokenStore {

    private static final String PREF_NAME = "pathlingo_auth";
    private static final String KEY_TOKEN = "sanctum_token";

    private final SharedPreferences prefs;

    public TokenStore(Context context) {
        prefs = context.getSharedPreferences(PREF_NAME, Context.MODE_PRIVATE);
    }

    public String getToken() {
        return prefs.getString(KEY_TOKEN, null);
    }

    public void save(String token) {
        prefs.edit().putString(KEY_TOKEN, token).apply();
    }

    public void clear() {
        prefs.edit().remove(KEY_TOKEN).apply();
    }
}

File: android/app/src/main/java/com/scholarlyapps/pathlingo/data/local/TokenStore.kt
package com.scholarlyapps.pathlingo.data.local

import android.content.Context
import androidx.datastore.core.DataStore
import androidx.datastore.preferences.core.Preferences
import androidx.datastore.preferences.core.edit
import androidx.datastore.preferences.core.stringPreferencesKey
import androidx.datastore.preferences.preferencesDataStore
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.first
import kotlinx.coroutines.flow.map

private val Context.dataStore: DataStore<Preferences> by preferencesDataStore("pathlingo_auth")

class TokenStore(private val context: Context) {

    val tokenFlow: Flow<String?> = context.dataStore.data.map { it[KEY_TOKEN] }

    suspend fun token(): String? = tokenFlow.first()

    suspend fun save(token: String) {
        context.dataStore.edit { it[KEY_TOKEN] = token }
    }

    suspend fun clear() {
        context.dataStore.edit { it.remove(KEY_TOKEN) }
    }

    companion object {
        private val KEY_TOKEN = stringPreferencesKey("sanctum_token")
    }
}
